persistence-modules/hibernate-annotations/src/main/java/com/baeldung/hibernate/customtypes/PhoneNumberType.java

```diff
@@ -10,7 +15,7 @@
 import java.sql.Types;
 import java.util.Objects;
 
-public class PhoneNumberType implements UserType<PhoneNumber> {
+public class PhoneNumberType implements UserType<PhoneNumber>, CompositeUserType<PhoneNumber> {
```

Comment:
it needs to implement the CompositeUserType interface since it has multiple columns